packages/redis/test/Adaptor.test.js

```diff
@@ -0,0 +1,334 @@
+    expect(result.data).to.eql('cat');
+  });
+  it('should return null if field value is not found', async () => {
+    setMockClient({
+      hGet: async () => null,
+    });
+    const state = {};
+    const result = await hget('animals:1', 'cat')(state);
+    expect(result.data).to.eql(null);
+  });
+  it('should throw field value is not specified', async () => {
+    setMockClient({
+      hGet: async () => {
+        throw new Error('TypeError: Invalid argument type');
+      },
+    });
+    const state = {};
+    try {
+      await hget('animals:1')(state);
+    } catch (error) {
+      expect(error.message).to.eql('TypeError: Invalid argument type');
+    }
+  });
+  it('should expand references', async () => {
+    setMockClient({
+      hGet: async () => 'cat',
+    });
+    const state = { key: 'animals:1', value: 'species' };
+    const result = await hget(
+      s => s.key,
+      s => s.species
+    )(state);
+    expect(result.data).to.eql('cat');
+  });
+});
+
+describe('set', () => {
+  it('should set a value for a key', async () => {
+    setMockClient({
+      set: async () => 'OK',
+    });
+
+    const state = {};
+    const result = await set('name', 'kwaku')(state);
+    expect(result.data).to.eql('OK');
+  });
+  it('should expand references', async () => {
+    setMockClient({
+      set: async () => 'OK',
+    });
+    const state = { key: 'name', value: 'mtuchi' };
+    const result = await set(
+      s => s.key,
+      s => s.value
+    )(state);
+    expect(result.data).to.eql('OK');
+  });
+  it('should throw an error if key and value is not specified', async () => {
+    setMockClient({
+      set: async () => {
+        throw new Error('TypeError: Invalid argument type');
+      },
+    });
+
+    const state = {};
+    try {
+      await set()(state);
+    } catch (error) {
+      expect(error.message).to.eql('TypeError: Invalid argument type');
+    }
+  });
+  it('should throw an error if value is not specified', async () => {
+    setMockClient({
+      set: async () => {
+        throw new Error('TypeError: Invalid argument type');
+      },
+    });
+
+    const state = {};
+    try {
+      await set('patient')(state);
+    } catch (error) {
+      expect(error.message).to.eql('TypeError: Invalid argument type');
+    }
+  });
+  it('should throw an error if key is not specified', async () => {
+    setMockClient({
+      set: async () => {
+        throw new Error('TypeError: Invalid argument type');
+      },
+    });
+
+    const state = {};
+    try {
+      await set('', 'kwaku')(state);
+    } catch (error) {
+      expect(error.message).to.eql('TypeError: Invalid argument type');
+    }
+  });
+});
+
+describe('hset', () => {
+  it('should set fields value for a specified key', async () => {
+    setMockClient({
+      hSet: async () => 1,
+    });
+
+    const state = {};
+    const result = await hset('patient', { name: 'kwaku' })(state);
```

Comment:
Ooh my bad 🙌🏽 , during implementation i i discovered two ways of using `client.hSet()` and i forgotten to update the notes to give you a heads up,

So there are two ways of setting hash values using `client.hSet()`

- Setup one field value at a time
> `client.hSet(key: string, field: string, value:string)`


- Setting up multiple field values 
> `client.hSet(key: string, fieldValues: object)`

I implemented the second signature because it most cases you would want to set the values in bulk like this 
```js
hset('noderedis:animals:1', {name: 'Fluffy', species: 'cat', age: 3}),
```
and not like this 👇🏽 
```
hset('key', 'a', '1')
or 
each($.mappings, hset($.key, $.field, $.value)
```
I am happy to change the implementation to use `hset(key, field, value)`.